features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Configuration/Configuration.jsx

```diff
@@ -355,84 +355,97 @@ class Configuration extends React.Component {
+                                                                    <FormattedMessage
+                                                                        id={'Apis.Details.Configuration.' +
+                                                                        'Configuration.default.version.tooltip'}
+                                                                        defaultMessage={'Marks one API version in a ' +
+                                                                        'group as the default so that it can be ' +
+                                                                        'invoked without specifying the version ' +
+                                                                        'number in the URL. For example, if you mark ' +
+                                                                        'http://host:port/youtube/2.0 as the default ' +
+                                                                        'API, requests made to http://host:port/' +
+                                                                        'youtube/ are automatically routed to version' +
+                                                                        ' 2.0. If you mark an unpublished API as the ' +
+                                                                        'default, the previous default published API ' +
+                                                                        'will still be used as the until the new ' +
+                                                                        'default API is published.'}
+                                                                    />
+                                                                </React.Fragment>
+                                                            }
+                                                        >
+                                                            <Button className={classes.helpButton}>
+                                                                <HelpOutline className={classes.helpIcon} />
+                                                            </Button>
+                                                        </Tooltip>
+                                                    </Typography>
+                                                    <Typography component='p' variant='body1'>
+                                                        <RadioGroup
+                                                            name='isDefaultVersion'
+                                                            className={classes.group}
+                                                            value={this.getDefaultVersion(
+                                                                isDefaultVersion,
+                                                                api.isDefaultVersion,
+                                                            )}
```

Comment:
won't this cause a re-render, as the function is called upon rendering?
Shouldn't this be as `onChange={() => this.handleChange('isDefaultVersion')}`